clearwater-diags-monitor/usr/share/clearwater/bin/gather_diags_utils

```diff
@@ -0,0 +1,95 @@
+#!/bin/sh
+
+# @file gather_diags_verification
+#
+# Copyright (C) Metaswitch Networks 2017
+# If license terms are provided to you in a COPYING file in the root directory
+# of the source code repository by which you are accessing this code, then
+# the license outlined in that COPYING file applies to your use.
+# Otherwise no rights are granted except for those provided to you by
+# Metaswitch Networks in a separate written agreement.
+
+
+# Functions that can be called to:
+# - Check the diags monitor process is actually running
+# - Get the current CPU idle percentage (rounded down)
+# - Check the user is happy to run a CPU intensive operation
+
+
+# Check for a diags monitor process. Alert, and attempt restart, if none found
+verify_process_running ()
+{
+  process_running=$(pgrep -f "/usr/share/clearwater/bin/clearwater_diags_monitor" > /dev/null ; echo $?)
+  if [ $process_running != 0 ]
+  then
+    echo "Warning: No clearwater-diags-monitor process found running. Attempting restart"
+    service clearwater-diags-monitor start
+    if [ $? != 0 ]
+    then
+      echo "Error: Unable to start clearwater-diags-monitor. Aborting operation"
+      exit 1
+    else
+      # Double check that the process stays up after the restart
+      sleep 5
+      process_running=$(pgrep -f "/usr/share/clearwater/bin/clearwater_diags_monitor" > /dev/null ; echo $?)
+      if [ $process_running != 0 ]
+      then
+        echo "Error: Clearwater-diags-monitor not running after restart. Aborting operation"
+        exit 1
+      fi
+      echo "Successfully started clearwater-diags monitor"
+    fi
+  fi
+}
+
+# Required idle CPU for triggering gathering diagnostics
+MIN_IDLE_CPU_FOR_GATHER=40
+
+# Gets the idle CPU averaged over a 10s period.
```

Comment:
Is 10s even remotely long enough? I'd have thought using loadavg across the last 1 minutes (see `/proc/loadavg`) would be more sensible.

Reply:
What it's always been, don't think it's worth shaking up at this point. Also, looking at https://linux.die.net/man/5/proc states:
```
/proc/loadavg
The first three fields in this file are load average figures giving the number of jobs in the run queue (state R) or waiting for disk I/O (state D) averaged over 1, 5, and 15 minutes. They are the same as the load average numbers given by uptime(1) and other programs. The fourth field consists of two numbers separated by a slash (/). The first of these is the number of currently runnable kernel scheduling entities (processes, threads). The value after the slash is the number of kernel scheduling entities that currently exist on the system. The fifth field is the PID of the process that was most recently created on the system.```

Not really sure how the number of jobs in queue or the rest would help us here